pkg/kubelet/cm/cgroupfs/qos_manager.go

```diff
@@ -0,0 +1,112 @@
+// +build linux
+
+/*
+Copyright 2016 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cgroupfs
+
+import (
+	"fmt"
+	"path"
+
+	"github.com/opencontainers/runc/libcontainer/cgroups"
+	"github.com/opencontainers/runc/libcontainer/cgroups/fs"
+	"github.com/opencontainers/runc/libcontainer/configs"
+	"k8s.io/kubernetes/pkg/kubelet/cm"
+)
+
+const (
+	defaultContainerRoot string = "/"
+)
+
+// qosContainerManagerImpl implements QOSContainerManager interface.
+// It internally uses libcontainer's cgroup manager implementation
+// for qos container creation, and updates.
+type qosContainerManagerImpl struct {
```

Comment:
The name of the file is qos_manager while the class is qosContainerManager, which seems like a mismatch.